spec/memoize_query_spec.rb

```diff
@@ -612,4 +537,246 @@ def expect_to_eq_with_or_without_redis
       expect_to_eq_with_or_without_redis { relation_with_and_or_clause.reload }
     end
   end
+
+  context 'when the queries have compare operators (unbound)' do
+    RSpec.shared_examples 'only memoizes when there are bound conditions' do
+      it 'does not memoize when there are only comparators' do
+        relations_with_only_comparator.each do |relation|
+          expect_not_to_use_redis do
+            expect(relation.to_a).to eq(result)
+          end
+        end
+      end
+
+      it 'memoizes queries with both comparator and other bound conditions' do
+        relations_with_comparator_and_other.each do |relation|
+          expect_to_use_redis do
+            expect(relation.to_a).to eq(result)
+          end
+        end
+      end
+    end
+
+    RSpec.shared_examples 'when update happens to the records' do
+      it 'it updates the affected query result' do
+        update
+        relations_with_comparator_and_other.each do |relation|
+          expect_to_use_redis do
+            expect(relation.reload.to_a).to eq(result)
+          end
+        end
+      end
+    end
+
+    RSpec.shared_examples 'with local cache' do
+      it 'only invalidates the affected query result sets' do
+        RedisMemo::Cache.with_local_cache do 
+          relations_with_comparator_and_other.each do |relation|
+            expect_to_use_redis do
+              expect(relation.to_a).to eq(unaffected_result)
+            end
+          end
+  
+          unaffected_creation
+  
+          # Does not affect with the creation because it is not dependent
+          relations_with_comparator_and_other.each do |relation|
+            expect_not_to_use_redis do
+              expect(relation.reload.to_a).to eq(unaffected_result)
+            end
+          end
+  
+          update
+  
+          # Will affect with the update
+          relations_with_comparator_and_other.each do |relation|
+            expect_to_use_redis do
+              expect(relation.reload.to_a).to eq(affected_result)
+            end
+          end
+        end
+      end
+    end
+  end
+
+  context 'when the queries have greater than (and or equal) operators' do
+    let!(:record) { Site.create!(a: 2, b: 2) }
+    
+    # The following two relations are both SELECT * FROM Site where a > 1
+    let!(:relation1_with_only_comparator) { Site.where('a > ?', 1) }
+    let!(:relation2_with_only_comparator) { Site.where(a: 2..Float::INFINITY) }
+    let!(:relation3_with_only_comparator) { Site.where('a > ? AND b = ?', 1, 2) }
```

Comment:
This will still be considered as only having comparators since it will be parsed as `Arel::Nodes::Grouping`, so will not be cached.